builder/builder.go

```diff
@@ -0,0 +1,210 @@
+package builder
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/ercole-io/ercole-agent/config"
+	"github.com/ercole-io/ercole-agent/marshal"
+	"github.com/ercole-io/ercole-agent/model"
+	"github.com/ercole-io/ercole-agent/utils"
+)
+
+func BuildData(configuration config.Configuration, version string, hostDataSchemaVersion int) *model.HostData {
+	hostData := new(model.HostData)
+
+	if configuration.Hostname != "default" {
+		hostData.Hostname = configuration.Hostname
+	}
```

Comment:
Are you sure that the hostdata.Hostname should be checked here and not after the line 29?

Reply:
You are surely right! I'm gonna fix this.